src/indexing.jl

```diff
@@ -48,6 +47,31 @@ end
     return v
 end
 
+# Slice
+@inline function Base.getindex{dim}(S::Tensor{2,dim}, i::Colon, j::Int)
+    @boundscheck checkbounds(S,i,j)
+    @inbounds v = Vec{dim}(S.data[(1+dim*(j-1)):dim*j])
+    return v
+end
+
+@inline function Base.getindex{dim}(S::Tensor{2,dim}, i::Int, j::Colon)
+    @boundscheck checkbounds(S,i,j)
+    @inbounds v = Vec{dim}(S.data[i:dim:(dim*(dim-1) + i)])
+    return v
+end
+
+@inline function Base.getindex{dim}(S::SymmetricTensor{2,dim}, i::Colon, j::Int)
+    @boundscheck checkbounds(S,i,j)
+    idx = Array(Int,dim)
```

Comment:
Creating this array will be expensive. We have to unroll the loop with a generated function.